services/QuillLMS/app/models/subscription.rb

```diff
@@ -152,6 +152,10 @@ def self.expired_today_or_previously_and_recurring
     Subscription.where('expiration <= ? AND recurring IS TRUE AND de_activated_date IS NULL', Date.today)
   end
 
+  def self.expired_today_or_previously_and_not_recurring
+    Subscription.where('expiration <= ? AND recurring IS FALSE AND de_activated_date IS NULL', Date.today)
```

Comment:
style nit: the latter two conditionals can be expressed in AREL syntax: `recurring: false...`